### DISCUSSION TITLE
Supabase asking for the optional `Vonage from` field as required in Phone Auth Provider Setup

### DISCUSSION THREAD

**Discussion by RayyanNafees** [supabase/supabase#16910] [Contribute to Supabase]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

### The [documentation](https://supabase.com/docs/guides/auth/phone-login/vonage#configure-supabase) says the field is optional

While Supabase [Phone Auth Provider Setup: https://supabase.com/dashboard/project/PROJECT-NAME/auth/providers](https://supabase.com/dashboard/project/PROJECT-NAME/auth/providers) asks for it as required field

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Go to [Auth Provider Setup: https://supabase.com/dashboard/project/PROJECT-NAME/auth/providers](https://supabase.com/dashboard/project/PROJECT-NAME/auth/providers)
2. Click on 'Phone'
3. Choose 'Vonage' as Verifying service
4. Scroll down to 'Vonage From' field
5. See the bug as it makes it required which contradicts the docs

## Expected behavior

The `Vonage From` Field should be optional as said in the [docs](https://supabase.com/docs/guides/auth/phone-login/vonage#configure-supabase)  on the Phone Auth Provider Setup or should be editted in the docs

## Screenshots

![image](https://github.com/supabase/supabase/assets/59685062/618bde13-3835-4e3e-812f-3a655605f256)

![image](https://github.com/supabase/supabase/assets/59685062/3b4c3c29-39a9-4a9a-9dfd-f265d5a605bb)

## System information

- OS: Windows
- Browser: chrome
- Version of supabase-js:  latest
- Version of Node.js: 20

**Comment by MildTomato** [comment]
What is the 'Vonage from' input referring to ?
cc @kangmingtay @J0 

There doesn't seem to be a phone number input on this form.

**Reply by J0** [reply]
I believe it should refer to the  number which is sending the SMS content.  In general, I think we should update the docs for Vonage. Created an internal task for htis

**Comment by dragonmantank** [comment]
Hello! I just caught wind of the fact that Vonage is in the Supabase project via a member of our community doing a video of it. I'm on the Vonage API Tooling team, let me know if there is any way I can help.

As a quick note, the whole "from" field is a bit nebulous because it's really country dependent. For example, the US requires the `FROM` number to be either a registered 10 digit number, or a toll free number (which also technically needs registration except for very low volumes). Many European countries require the `FROM` to be an alphanumeric string that may or may not need to be registered with the country's carriers.

There are also potentially better APIs this can use instead of our base SMS API that will get around a lot of the above issues.

**Reply by kangmingtay** [reply]
hey @dragonmantank , thanks for letting us know about this! seems like the `from` field is required in most cases here - are there any exceptions where you would not want the `from` field set?

**Reply by dragonmantank** [reply]
No, `from` will always be required. It just won't always be a numeric value, as that depends on the country.
